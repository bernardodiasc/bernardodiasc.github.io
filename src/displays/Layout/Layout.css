.Layout {
  display: flex;
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
  font-size: 1rem;
}

.Layout__content {
  flex: 1;
  max-width: 800px;
  background: rgba(245, 245, 245, 0.9);
  padding: 2em 2em 3em;
  margin: 1em 0 8em;
  border-radius: 8px;
  box-shadow: 0 0 100px rgba(45, 45, 45, 0.3);
}

.Layout__headerbar {
  width: 300px;
  padding: 3em 1em;
}
