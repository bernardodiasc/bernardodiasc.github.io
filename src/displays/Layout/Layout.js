import React, { PureComponent } from 'react'
import './Layout.css'

import HeaderBar from './HeaderBar'
import Languages from './Languages'

class Layout extends PureComponent {
  static defaultProps = {
    children: null,
    breadcrumbs: [],
    languages: {},
  }

  render() {
    return (
      <main className="Layout">
        <div className="Layout__headerbar">
          <HeaderBar languages={<Languages {...this.props.languages} />} />
        </div>
        <article className="Layout__content">
          {this.props.children}
        </article>
      </main>
    )
  }
}

export default Layout
