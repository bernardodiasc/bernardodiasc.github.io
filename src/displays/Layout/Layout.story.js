import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import Docs from 'stories/containers/Docs'

import Layout from './Layout'
import LayoutReadme from './README.md'
storiesOf('Displays/Layout', module)
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{LayoutReadme}</Docs>)
  .add('Example', () => <Layout />)
  .add('Minimal render', () => <Layout />)
