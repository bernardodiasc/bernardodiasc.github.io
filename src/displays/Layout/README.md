# Layout

## How to use

```
import Layout from 'displays/Layout'
```

```
<Layout />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
