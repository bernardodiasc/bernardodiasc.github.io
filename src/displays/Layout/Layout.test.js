import React from 'react'
import { shallow } from 'enzyme'

import Layout from './Layout.js'

it('renders without crashing', () => {
  const component = shallow(<Layout />)
  expect(component).toMatchSnapshot()
})
