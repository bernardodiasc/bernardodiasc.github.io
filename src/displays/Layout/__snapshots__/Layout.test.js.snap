// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<main
  className="Layout"
>
  <div
    className="Layout__headerbar"
  >
    <HeaderBar
      languages={
        <Languages
          activeLanguage={Array []}
          availableLanguages=""
          selectLanguage={[Function]}
        />
      }
    />
  </div>
  <article
    className="Layout__content"
  />
</main>
`;
